x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -626,6 +628,15 @@ private static boolean checkGroupMatch(Expression e, Node<?> source, List<Expres
         return false;
     }
 
+    private static void checkBooleanFiltering(LogicalPlan p, Set<Failure> localFailures) {
```

Comment:
Filtering only works on bool hence why I recommend renaming it to `checkFilterConditionType`